packages/effect/src/Cron.ts

```diff
@@ -276,58 +287,76 @@ export const next = (cron: Cron, now?: Date): Date => {
   const restrictMonths = months.size !== 0
   const restrictWeekdays = weekdays.size !== 0
 
-  const current = now ? new Date(now.getTime()) : new Date()
-  // Increment by one minute to ensure we don't match the current date.
-  current.setMinutes(current.getMinutes() + 1)
-  current.setSeconds(0)
-  current.setMilliseconds(0)
-
-  // Only search 8 years into the future.
-  const limit = new Date(current).setFullYear(current.getFullYear() + 8)
-  while (current.getTime() <= limit) {
-    if (restrictMonths && !months.has(current.getMonth() + 1)) {
-      current.setMonth(current.getMonth() + 1)
-      current.setDate(1)
-      current.setHours(0)
-      current.setMinutes(0)
+  // TODO: This is unsafe.
+  const zoned = dateTime.unsafeMakeZoned(now ?? new Date())
+  const adjusted = Option.isSome(cron.tz) ? dateTime.setZone(zoned, cron.tz.value) : zoned
+
+  let current = dateTime.setParts(adjusted, {
+    // Increment by one minute to ensure we don't match the current date.
+    minutes: dateTime.getPart(adjusted, "minutes") + 1,
+    seconds: 0,
+    millis: 0
+  })
+
+  // TODO: This algorithm can be optimized to avoid some unnecessary iterations.
+  for (let i = 0; i < 10_000; i++) {
+    const parts = dateTime.toParts(current)
+
+    if (restrictMonths && !months.has(parts.month)) {
+      current = dateTime.setParts(current, {
+        month: dateTime.getPart(current, "month") + 1,
+        day: 1,
+        hours: 0,
+        minutes: 0
+      })
```

Comment:
I have not profiled this but it's probably not very efficient compared to the previous mutative implementation. Maybe it's possible to implement a version of `DateTime.mutate` that gives access to the actual `DateTime` object instead of the internal `Date` representation.